src/sys/redox/mod.rs

```diff
@@ -0,0 +1,31 @@
+mod awakener;
+mod eventedfd;
+mod io;
+mod selector;
+mod tcp;
+mod udp;
+
+pub use self::awakener::Awakener;
+pub use self::eventedfd::EventedFd;
+pub use self::io::{Io, set_nonblock};
+pub use self::selector::{Events, Selector};
+pub use self::tcp::{TcpStream, TcpListener};
+pub use self::udp::UdpSocket;
+
+use std::os::unix::io::FromRawFd;
+
+pub fn pipe() -> ::io::Result<(Io, Io)> {
+    use syscall::{O_NONBLOCK, O_CLOEXEC};
+    use syscall::pipe2;
+
+    let mut fds = [0; 2];
```

Comment:
Maybe use `-1` so the code can't even mess with fd 0. Syscalls on fd `-1` should returns errors (right?).

Reply:
Ah, didn't know that.